sql: mark implicit crdb_region column as HIDDEN

otan opened the pull request:
Last commit is relevant, other commits are from PR(s) #59828 

Resolves #59630 

----

Release note (sql change): REGIONAL BY ROW tables which have an
implicitly created crdb_region table will now mark the given column as
hidden so it does not display in `SELECT *` statements.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59831)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
This one <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>.  Reinforces my comment on #59828 though, that we should test roundtrip-ability.

> Reviewed 38 of 38 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59831#-:-MSnlTql2btLZU98HYxe:b-ntew1d)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajstorm and @dt)



<!-- Sent from Reviewable.io -->

otan committed:
sql: mark implicit crdb_region column as HIDDEN

Release note (sql change): REGIONAL BY ROW tables which have an
implicitly created crdb_region table will now mark the given column as
hidden so it does not display in `SELECT *` statements.

otan commented:
bors r=ajstorm
